all/pom.xml

```diff
@@ -36,11 +36,35 @@
     </parent>
 
     <dependencies>
+        <!-- this is ordered in a special way so that where nested dependencies bring in different versions
```

Comment:
Wow, this feels subtle (and brittle for when dependencies are next updated). Is it possible to instead use `<excludes>` or some such, so that we're only trying to pull in the one version of each from brooklyn?

Reply:
there might be a lot of complex excludes to set up (and maintenance hell in the longer term).  if it becomes really problematic then that's something to consider but for now i think best to put more dependencies in the pom wherever we need to prioritize some versions over others.